examples/plot_intersectional_bias.py

```diff
@@ -0,0 +1,1327 @@
+#   this problem through an intersectional lens
+#
+# We can take a look at the data to get an idea about what might be going on. First, we examine the
+# frequency of diagnoses, stratified by intersectional groups.
+
+# %%
+
+m = sns.FacetGrid(train, row="Sex", col="Race")
+m.map(sns.histplot, "Diagnosis", discrete=True, shrink=0.8)
+
+# %%
+# We see that the frequency of the two diagnoses (of affective disorder or 0 and schizophrenia
+# or 1) are fairly similar among all intersectional subgroups, except Black men, who have a much
+# higher rate of schizophrenia diagnosis.
+#
+# Fairness through Unawareness?
+# =============================
+#
+# In our data, Black men are less likely to be diagnosed with affective disorder, and more likely
+# to be diagnosed with schizophrenia than other groups (and almost two times as likely as white
+# men). Perhaps the model is picking up on this trend, which is contributing to bias. If we remove
+# race, the classifier will no longer have access to this information during training.
+#
+# This is an approach that is commonly termed **fairness through unawareness.** Specifically, it
+# refers to the notion that a model that does not have access to a given feature when making
+# predictions cannot be unfair with respect to that feature.
+
+# %%
+# Define and drop race-related variables
+race_cat = ["Race_Asian", "Race_Black", "Race_Hispanic", "Race_White"]
+trainx_norace = trainx.drop(race_cat, axis=1)
+testx_norace = testx.drop(race_cat, axis=1)
+
+# %%
+# Now that we've dropped the variables, we'll train a second model that is identical to the
+# first one, except it no longer uses Race as a feature.
+
+# %%
+
+# Define and train a second model
+model2 = LogisticRegression(penalty="elasticnet", max_iter=1000, solver="saga", l1_ratio=1)
+model2 = model2.fit(trainx_norace, trainy)
+
+train_predictions = model2.predict(trainx_norace)
+print("Training accuracy: ", skm.accuracy_score(trainy, train_predictions))  # Training accuracy
+
+predictions = model2.predict(testx_norace)
+print("Test accuracy: ", skm.accuracy_score(testy, predictions))  # Test accuracy
+
+# %%
+# We'll note that the accuracy is still solid. It has decreased slightly, which makes sense
+# given that our model has access to fewer features with which to maximize predictive accuracy.
+#
+# That said, our most important objective is to analyze whether fairness has increased as a result
+# of our change, so let's perform another bias assessment.
+
+# %%
+f(testy, predictions)
+
+# %%
+# Oh yikes - by removing race from our model, the FPR for the Black population has increased to
+# 3.8x that of our reference group.
+#
+# Does intersectional fairness show a similar trend?
+
+# %%
+
+intersectionalf(testy, predictions)
+
+# %%
+# If we analyze this problem through an intersectional lens, we notice that the relative FPR has
+# increased even further, and Black men are now being misdiagnosed at a rate that is 7x higher than
+# white men.
```

Comment:
Suggest mentioning that proxy variables can be highly correlated with sensitive attributes like race, therefore removing race alone may not always be enough to remove its historical impact on the data-generating process.